src/cclib/parser/daltonparser.py

```diff
@@ -737,20 +754,36 @@ def extract(self, inputfile, line):
         # @    Final gradient norm:           0.000003746706
         # ...
         #
+        if "Final HF energy" in line and not (hasattr(self, "mpenergies") or hasattr(self, "ccenergies")):
+            self.metadata["theory"] = "HF"
+        if "Final DFT energy" in line:
+            self.metadata["theory"] = "DFT"
+        if "This is a DFT calculation of type" in line:
+            self.metadata["functional"] = line.split()[-1]
+
         if "Final DFT energy" in line or "Final HF energy" in line:
             if not hasattr(self, "scfenergies"):
                 self.scfenergies = []
             temp = line.split()
             self.scfenergies.append(utils.convertor(float(temp[-1]), "hartree", "eV"))
 
         if "@   = MP2 second order energy" in line:
+            self.metadata["theory"] = "MP2"
             energ = utils.convertor(float(line.split()[-1]), 'hartree', 'eV')
             if not hasattr(self, "mpenergies"):
                 self.mpenergies = []
             self.mpenergies.append([])
             self.mpenergies[-1].append(energ)
 
+        if "Total CC2   energy:" in line:
+            self.metadata["theory"] = "CCSD"
+            energ = utils.convertor(float(line.split()[-1]), 'hartree', 'eV')
+            if not hasattr(self, "ccenergies"):
+                self.ccenergies = []
+            self.ccenergies.append(energ)
+
         if "Total energy CCSD(T)" in line:
+            self.metadata["theory"] = "CCSD-T"
```

Comment:
CSX is in XML format now, which doesn't allow parentheses in the element. I know it's annoying. In the future, CSX might use JSON-LD serialization. Then we can CCSD(T)